integration-tests/emulated/assets/asset-hub-kusama/src/tests/hrmp_channels.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::*;
+
+const MAX_CAPACITY: u32 = 8;
+const MAX_MESSAGE_SIZE: u32 = 8192;
+
+/// Opening HRMP channels between Parachains should work
+#[test]
+fn open_hrmp_channel_between_paras_works() {
+	// Parchain A init values
+	let para_a_id = PenpalKusamaA::para_id();
+	let para_a_root_origin = <PenpalKusamaA as Chain>::RuntimeOrigin::root();
+
+	// Parachain B init values
+	let para_b_id = PenpalKusamaB::para_id();
+	let para_b_root_origin = <PenpalKusamaB as Chain>::RuntimeOrigin::root();
+
+	let fee_amount = KUSAMA_ED * 1000;
+	let fund_amount = KUSAMA_ED * 1000_000_000;
+
+	// Fund Parachain's Sovereign accounts to be able to reserve the deposit
+	let para_a_sovereign_account = Kusama::fund_para_sovereign(fund_amount, para_a_id);
+	let para_b_sovereign_account = Kusama::fund_para_sovereign(fund_amount, para_b_id);
+
+	let relay_destination: VersionedMultiLocation = PenpalKusamaA::parent_location().into();
+
+	// ---- Init Open channel from Parachain to System Parachain
+	let mut call = Kusama::init_open_channel_call(para_b_id, MAX_CAPACITY, MAX_MESSAGE_SIZE);
+	let origin_kind = OriginKind::Native;
+	let native_asset: MultiAsset = (Here, fee_amount).into();
+	let beneficiary = Kusama::sovereign_account_id_of_child_para(para_a_id);
+
+	let mut xcm = xcm_transact_paid_execution(call, origin_kind, native_asset.clone(), beneficiary);
+
+	PenpalKusamaA::execute_with(|| {
+		assert_ok!(<PenpalKusamaA as PenpalKusamaAPallet>::PolkadotXcm::send(
+			para_a_root_origin,
+			bx!(relay_destination.clone()),
+			bx!(xcm),
+		));
+
+		PenpalKusamaA::assert_xcm_pallet_sent();
+	});
+
+	Kusama::execute_with(|| {
+		type RuntimeEvent = <Kusama as Chain>::RuntimeEvent;
+
+		Kusama::assert_ump_queue_processed(
+			true,
+			Some(para_a_id),
+			Some(Weight::from_parts(1_312_558_000, 200000)),
```

Comment:
nit: those hardcoded numbers are going to be a maintenance nightmare.
The way my Chopsticks e2e tests doing is round the numbers so small adjustment between versions won't impact the test outcome, and use snapshots testing tool to simply the process of updating them.

Reply:
all of these `Option<Weight>` params used on events asserts [are using `weight_within_threshold` under the hood](https://github.com/acatangiu/runtimes/blob/77da564b3b9bfe6a88642c8fba3d55dccf6a825f/integration-tests/emulated/common/src/impls.rs#L271), where thresholds [currently allow 33% delta](https://github.com/acatangiu/runtimes/blob/13fe2f99dd10823941a59f5ccda78b19d0980cb4/integration-tests/emulated/common/src/constants.rs#L41).

Everybody (including myself) initially thinks it's checking exact weights because of the very specific numbers used. Maybe rounding them up like you suggested is a good idea to hint to the fact they are actually estimates/thresholds.